### PR TITLE
Implement UNICODE function

### PR DESCRIPTION
**Pull request by ExtraBB**
First contribution, thought I'd start with something simple. Let me know if I missed anything.

### CODE REVIEW

**Comment by nhatcher** [comment]
This is great!!

For what I can see you are just missing a test workbook. You just need to create a workbook with Excel called UNICODE.xlsx and place it here:

https://github.com/ironcalc/IronCalc/tree/main/xlsx/tests/calc_tests

The heavy lifting of the tests should be in that file

**Review by nhatcher — CHANGES_REQUESTED** [review]
Missing test workbook

**Comment by ExtraBB** [comment]
Thanks, I completely missed the `xlsx` tests but they are very useful! Caught a bug writing to the wrong Excel formatting and missing the `_xlfn` variant of the function. The branch is updated

**Comment by nhatcher** [comment]
Hi @ExtraBB , don't worry about the test coverage.
I will get around doing a proper review on my evening (Berlin time)

**Comment by nhatcher** [comment]
Awesome, sorry for the late reply. The test function is great but doesn't contain error cases:

image

Also can you color those cells that have a formula:

image

Then it is easy to see for instance that A20 has a formula (as opposed to be empty)

**Comment by nhatcher** [comment]
You also need to add `UNICODE` here:

https://github.com/ironcalc/IronCalc/blob/main/base/src/functions/mod.rs#L452-L481

**Comment by ExtraBB** [comment]
@nhatcher updated with error cases.

File: base/src/test/test_fn_unicode.rs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    assert_eq!(model._get_text("A3"), *"#NAME?");
+    assert_eq!(model._get_text("A4"), *"#VALUE!");
+    assert_eq!(model._get_text("A5"), *"#REF!");
+    assert_eq!(model._get_text("A6"), *"#DIV/0!");
+}
+
+#[test]
+fn wrong_number_of_arguments() {
+    let mut model = new_empty_model();
+    model._set("A1", "=UNICODE()");
```
Comment (nhatcher) [inline base/src/test/test_fn_unicode.rs:57]: One last thing, two arguments?

**Review by nhatcher — CHANGES_REQUESTED** [review]

File: base/src/test/test_fn_unicode.rs (hunk 1 above)
Comment (exbb123bb) [reply base/src/test/test_fn_unicode.rs:57]: Updated 👍

File: base/src/functions/text.rs [hunk 2]
```diff
@@ -342,6 +342,50 @@ impl Model {
...
+
+            if s.len() == 0 {
+                return CalcResult::Error {
+                    error: Error::VALUE,
+                    origin: cell,
+                    message: "Empty cell".to_string(),
+                };
+            }
+
+            let unicode_number = s.chars().nth(0).unwrap() as u32;
```
Comment (nhatcher) [inline base/src/functions/text.rs]: Sorry, I missed this. We can't use `unwrap` (or `panic`) in the library code.

Comment (nhatcher) [reply base/src/functions/text.rs]: Maybe something like:

```Rust
match s.chars().next() {
    Some(c) => {
        return CalcResult::Number(c as f64);
    },
    None => {
        return CalcResult::Error {
            error: Error::VALUE,
            origin: cell,
            message: "Empty cell".to_string(),
        };
    }
}
```

**Review by nhatcher — CHANGES_REQUESTED** [review]
Sorry, I missed the unwrap.

I will add this to clippy eventually to make it impossible soon enough.

File: base/src/functions/text.rs (hunk 2 above)
Comment (exbb123bb) [reply base/src/functions/text.rs]: Updated! Can't cast `char` directly to `f64`, has to go through `u32` first

**Comment by nhatcher** [comment]
Approved and merged!

**Pull request merged**